p2test.py

```diff
@@ -143,6 +148,12 @@ def readLine(self, line):
 
         elif actor.startswith('Elf'):
             ID = int(actor.split()[1])
+
+            if self.bonus:
```

Comment:
If bonus is tested new elves will be generated here when its ID not found in array